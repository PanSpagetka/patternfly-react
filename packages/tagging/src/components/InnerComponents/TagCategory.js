import React from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip } from 'patternfly-react';
import Tag from './Tag';

class TagCategory extends React.Component {
  generateTag = tagValue => (
    <Tag
      key={tagValue.id}
      tagCategory={this.props.tagCategory}
      tagValue={tagValue}
      onTagDeleteClick={this.props.onTagDeleteClick}
      truncate={this.props.valueTruncate}
    />
  );

  render() {
    const values = [...this.props.tagValues];
    const categoryTooltip = (
      <Tooltip id="tooltip">{this.props.tagCategory.description}</Tooltip>
    );
    return (
      <ul className="tag-category list-inline">
        <OverlayTrigger placement="bottom" overlay={categoryTooltip}>
          <div className="category-label">
            {this.props.categoryTruncate(this.props.tagCategory.description)}
          </div>
        </OverlayTrigger>
        {values
          .sort((a, b) => (a.description < b.description ? -1 : 1))
          .map(tagValue => this.generateTag(tagValue))}
      </ul>
    );
  }
}

TagCategory.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: PropTypes.object.isRequired,
  tagValues: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  categoryTruncate: PropTypes.func,
  valueTruncate: PropTypes.func
};

TagCategory.defaultProps = {
  categoryTruncate: str =>
    str.length > 18 ? `${str.substring(0, 18)}...` : str,
  valueTruncate: str => (str.length > 18 ? `${str.substring(0, 18)}...` : str)
};

export default TagCategory;
