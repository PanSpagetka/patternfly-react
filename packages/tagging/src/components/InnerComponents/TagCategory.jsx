import React from 'react';
import PropTypes from 'prop-types';
import Tag from './Tag';

class TagCategory extends React.Component {

  generateTag = tagValue =>
    (<Tag
      key={tagValue.id}
      tagCategory={this.props.tagCategory}
      tagValue={tagValue}
      onTagDeleteClick={this.props.onTagDeleteClick}
      truncate={this.props.valueTruncate}
    />);

  render() {
    let values = [...this.props.tagValues];
    return (
      <ul className="tag-category list-inline">
        <div className="category-label" title={this.props.tagCategory.description}>{this.props.categoryTruncate(this.props.tagCategory.description)}</div>
        {values.sort((a,b) => a.description < b.description ? -1 : 1).map(tagValue => this.generateTag(tagValue))}
      </ul>
    );
  }
}

TagCategory.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: PropTypes.object.isRequired,
  tagValues: PropTypes.arrayOf(PropTypes.object).isRequired,
  categoryTruncate: PropTypes.func,
  valueTruncate: PropTypes.func,
};

TagCategory.defaultProps = {
  categoryTruncate: str => ( str.length > 18 ? str.substring(0, 18) + '...' : str ),
  valueTruncate: str => ( str.length > 18 ? str.substring(0, 18) + '...' : str ),
}

export default TagCategory;
