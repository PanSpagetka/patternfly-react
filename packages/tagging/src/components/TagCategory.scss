@import '~patternfly/dist/sass/patternfly/variables';

ul.tag-category.list-inline {
  background-color: $color-pf-blue-500;
  padding: 3px 10px 3px 15px !important;
  color: $color-pf-white;
  margin-left: 5px;
  margin-right: 10px;
  margin-bottom: 5px;
  font-size: 14px;
  line-height: 24px;
  display: table;
  float: left;
  li a span {
    color: $color-pf-white;
  }
}

.category-label {
  padding-right: 15px;
}
