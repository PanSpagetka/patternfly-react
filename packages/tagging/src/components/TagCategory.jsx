import React from 'react';
import PropTypes from 'prop-types';
import Tag from './Tag';
import './TagCategory.scss';

class TagCategory extends React.Component {
  generateTag = tagValue =>
    (<Tag
      key={tagValue.id}
      tagCategory={this.props.tagCategory}
      tagValue={tagValue}
      onTagDeleteClick={this.props.onTagDeleteClick}
    />);

  render() {
    return (
      <ul className="tag-category list-inline">
        <div className="category-label">{this.props.tagCategory.description}</div>
        {this.props.tagValues.map(tagValue => this.generateTag(tagValue))}
      </ul>
    );
  }
}

TagCategory.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: PropTypes.object.isRequired,
  tagValues: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default TagCategory;
