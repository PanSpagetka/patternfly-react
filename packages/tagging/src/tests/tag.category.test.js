import React from 'react';
import TagCategory from '../components/InnerComponents/TagCategory';
import renderer from 'react-test-renderer';

const tagCategory = { description: 'animal', id: 1 };
const tagValues = [
  { description: 'duck', id: 1 },
  { description: 'lion', id: 2 }
];
function onDelete(x) {
  return x;
}

describe('TagCategory Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(
      <TagCategory
        tagCategory={tagCategory}
        tagValues={tagValues}
        onTagDeleteClick={onDelete}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
