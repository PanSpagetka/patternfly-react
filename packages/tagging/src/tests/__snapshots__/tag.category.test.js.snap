// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCategory Component match snapshot 1`] = `
<ul
  className="tag-category list-inline"
>
  <div
    className="category-label"
  >
    animal
  </div>
  <li
    className="tag"
  >
    <span
      className="label label-primary"
      style={
        Object {
          "color": "defaultColor",
        }
      }
    >
      duck
      <a
        className="pf-remove-button"
        href="#"
        onClick={[Function]}
      >
        <span
          aria-hidden="true"
          className="pficon pficon-close"
        />
        <span
          className="sr-only"
        >
          Remove
        </span>
      </a>
    </span>
  </li>
  <li
    className="tag"
  >
    <span
      className="label label-primary"
      style={
        Object {
          "color": "defaultColor",
        }
      }
    >
      lion
      <a
        className="pf-remove-button"
        href="#"
        onClick={[Function]}
      >
        <span
          aria-hidden="true"
          className="pficon pficon-close"
        />
        <span
          className="sr-only"
        >
          Remove
        </span>
      </a>
    </span>
  </li>
</ul>
`;
